avro4s-core/src/test/scala/com/sksamuel/avro4s/record/decoder/FixedDecoderTest.scala

```diff
@@ -20,7 +20,7 @@ class FixedDecoderTest extends FunSuite with Matchers {
   test("support options of fixed") {
     val schema = AvroSchema[OptionalFixedValueType]
     val record = new GenericData.Record(schema)
-    record.put("z", Array[Byte](115, 97, 109))
+    record.put("z", new GenericData.Fixed(AvroSchema[FixedValueType], Array[Byte](115, 97, 109)))
```

Comment:
Should we keep both tests since Fixed doesn't always have to be GenericData.Fixed ?

Reply:
It doesn't? I didn't realise that :P. I will go and change that if you like so that the decoder can handle both?

Reply:
Yeah because people are allowed to make records themselves, and a normal array is fine (although GenericData.Fixed is what Avro itself would use when marshalling).